flink-runtime-web/src/main/java/org/apache/flink/runtime/webmonitor/handlers/JarRunHandler.java

```diff
@@ -114,17 +114,12 @@ public JarRunHandler(
 
 		CompletableFuture<JobGraph> jarUploadFuture = jobGraphFuture.thenCombine(blobServerPortFuture, (jobGraph, blobServerPort) -> {
 			final InetSocketAddress address = new InetSocketAddress(getDispatcherHost(gateway), blobServerPort);
-			final List<PermanentBlobKey> keys;
-			try {
-				keys = BlobClient.uploadFiles(address, configuration, jobGraph.getJobID(), jobGraph.getUserJars());
+			try (BlobClient blobClient = new BlobClient(address, configuration)) {
+				ClientUtils.uploadAndSetUserJars(jobGraph, blobClient);
```

Comment:
we _could_ use `uploadJobGraphFiles` here, but there isn't really a use-case for uploading distributed cache artifacts when going through the JarRunHandler, since we're already on the server here.

Reply:
But it would reduce code redundancy, right? If this is the case, then let's do it.

Reply:
The `JarRunHandler` now also uses `uploadJobGraphFiles()`.